src/dmd/link.d

```diff
@@ -1484,4 +1475,87 @@ version (Windows)
             }
         }
     }
+
+    ///////////////////////////////////////////////////////////////////////
+    // COM interfaces to find VS2017+ installations
+    import core.sys.windows.com;
+    import core.sys.windows.wtypes : BSTR;
+    import core.sys.windows.winnls : WideCharToMultiByte, CP_UTF8;
+    import core.sys.windows.oleauto : SysFreeString;
+
+    pragma(lib, "ole32.lib");
+    pragma(lib, "oleaut32.lib");
+
+    interface ISetupInstance : IUnknown
+    {
+        // static const GUID iid = uuid("B41463C3-8866-43B5-BC33-2B0676F7F42E");
+        static const GUID iid = { 0xB41463C3, 0x8866, 0x43B5, [ 0xBC, 0x33, 0x2B, 0x06, 0x76, 0xF7, 0xF4, 0x2E ] };
+
+        int GetInstanceId(BSTR* pbstrInstanceId);
+        int GetInstallDate(LPFILETIME pInstallDate);
+        int GetInstallationName(BSTR* pbstrInstallationName);
+        int GetInstallationPath(BSTR* pbstrInstallationPath);
+        int GetInstallationVersion(BSTR* pbstrInstallationVersion);
+        int GetDisplayName(LCID lcid, BSTR* pbstrDisplayName);
+        int GetDescription(LCID lcid, BSTR* pbstrDescription);
+        int ResolvePath(LPCOLESTR pwszRelativePath, BSTR* pbstrAbsolutePath);
+    }
+
+    interface IEnumSetupInstances : IUnknown
+    {
+        // static const GUID iid = uuid("6380BCFF-41D3-4B2E-8B2E-BF8A6810C848");
```

Comment:
This interface doesn't need an `iid`?

Reply:
No, it's returned directly by `ISetupConfiguration.EnumInstances`. `ISetupInstance.iid` is also not needed, but I had already converted it before I noticed.